k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolderPusher.java

```diff
@@ -359,16 +330,12 @@ private void checkConnectionIdleCapable(ImapConnection conn) throws MessagingExc
         }
 
         private void setReadTimeoutForIdle(ImapConnection conn) throws SocketException {
-            int idleRefreshTimeout = store.getStoreConfig().getIdleRefreshMinutes() * 60 * 1000;
+            int idleRefreshTimeout = folder.getStore().getStoreConfig().getIdleRefreshMinutes() * 60 * 1000;
             conn.setReadTimeout(idleRefreshTimeout + IDLE_READ_TIMEOUT_INCREMENT);
         }
 
         @Override
-        public void handleAsyncUntaggedResponse(ImapResponse response) {
-            if (K9MailLib.isDebug()) {
-                Timber.v("Got async response: %s", response);
-            }
-
+        public void handleAsyncUntaggedResponse(ImapResponse response) throws IOException, MessagingException {
```

Comment:
I think it is async in the sense that `executeSimpleCommand(Commands.IDLE, untaggedhandler)` is a  long blocking call, and while it is blocking, `handleAsyncUntaggedResponse` is called to process a response.